File: revision/15s2/hideemails.pl
#!/usr/local/bin/perl -w

open F, "<", $ARGV[0] or die "cannot open file";
foreach $line (<F>) {
    chomp $line;
    $line =~ s/ +/ /g;
    $line =~ s/\@[^ ]*/\@example.com/g;
    push @lines, $line;
}
close F;

open G, ">", $ARGV[0] or die "cannot write to file";
foreach $line (@lines) {
    print G $line, "\n";
}
close G;

File: revision/15s2/hideemails.py
#!/usr/bin/python3

import sys, re;

read_file = open(sys.argv[1])
file_lines = read_file.readlines()
lines = []

for line in file_lines:
    line = line.strip()
    line = re.sub(' +', ' ', line)
    line = re.sub(r'\@[^ ]*', '@example.com', line)
    lines.append(line)


write_file = open(sys.argv[1], "w")
for line in lines:
    write_file.write(line+"\n")
